apps/src/lib/util/GoogleAnalyticsReporter.js

```diff
@@ -0,0 +1,152 @@
+window.dataLayer = window.dataLayer || [];
+
+/**
+ * Global function used to track Google Universal events.
+ * It is a wrapper class that appends information to window.dataLayer
+ * This is the equivalent of ga() in Google Universal
+ */
+function gtag() {
+  window.dataLayer.push(arguments);
+}
+
+/**
+ * Class used to control the initialization of Google analytics
+ * for code.org properties.
+ * The class is responsible for
+ * - Adding the Google Tags needed by Tag manager
+ * - Send any dimensions information
+ * - Wrap Google Universal and Analytics functions for easier transition
+ * - Centralize common code
+ */
+class GoogleAnalyticsReporter {
+  enableGoogleAnalyticsUniversal = true;
+  enabledGoogleAnalytics4 = true;
+  googleUniversalGtag = 'UA-37745279-1';
+  googleAnalyticsGTag = 'G-L9HT5MZ3HD';
+
+  constructor() {
+    this.#initializeGoogleUniversal(
+      window,
+      window.document,
+      'script',
+      '//www.google-analytics.com/analytics.js',
+      'ga'
+    );
+
+    this.#initializeGoogleAnalytics();
+
+    this.#addEventListener();
+
+    window.trackEvent = this.trackEvent;
+    window.readCookie = this.readCookie;
+  }
+  /**
+   * Enable Google Universal. This adds the tags and the ga function to the window
+   * to be used across the site.
```

Comment:
Regarding @breville's comment about citing the source of this pasted code.

```suggestion
   * Enable Google Universal (deprecated 2023-07-01). This boilerplate code from
   * Google Analytics adds the tags and the `ga` function to the window to be used
   * across the site.
```